client/my-sites/email/email-management/home/email-plan-mailboxes-list.jsx

```diff
@@ -136,7 +159,125 @@ const getActionsForMailbox = ( mailbox, translate, dispatch ) => {
 	};
 };
 
-function EmailPlanMailboxesList( { accountType, mailboxes, isLoadingEmails } ) {
+const getTitanMenuItems = ( translate ) => {
+	return [
+		{
+			href: 'https://wp.titan.email/mail/',
+			image: titanMailIcon,
+			imageAltText: translate( 'Titan Mail icon' ),
+			title: translate( 'Mail', {
+				comment: 'This refers to the Email application (i.e. the webmail) of Titan',
+			} ),
+		},
+		{
+			href: 'https://wp.titan.email/calendar/',
+			image: titanCalendarIcon,
+			imageAltText: translate( 'Titan Calendar icon' ),
+			title: translate( 'Calendar', {
+				comment: 'This refers to the Calendar application of Titan',
+			} ),
+		},
+		{
+			href: 'https://wp.titan.email/contacts/',
+			image: titanContactsIcon,
+			imageAltText: translate( 'Titan Contacts icon' ),
+			title: translate( 'Contacts', {
+				comment: 'This refers to the Contacts application of Titan',
+			} ),
+		},
+	];
+};
+
+const getGmailMenuItems = ( domainName, translate ) => {
+	return [
+		{
+			href: getGmailUrl( domainName ),
+			image: gmailIcon,
+			imageAltText: translate( 'Gmail icon' ),
+			title: translate( 'View Gmail' ),
+		},
+		{
+			href: getGoogleAdminUrl( domainName ),
+			image: googleAdminIcon,
+			imageAltText: translate( 'Google Admin icon' ),
+			title: translate( 'View Admin' ),
+		},
+		{
+			href: getGoogleCalendarUrl( domainName ),
+			image: googleCalendarIcon,
+			imageAltText: translate( 'Google Calendar icon' ),
+			title: translate( 'View Calendar' ),
+		},
+		{
+			href: getGoogleDocsUrl( domainName ),
+			image: googleDocsIcon,
+			imageAltText: translate( 'Google Docs icon' ),
+			title: translate( 'View Docs' ),
+		},
+		{
+			href: getGoogleDriveUrl( domainName ),
+			image: googleDriveIcon,
+			imageAltText: translate( 'Google Drive icon' ),
+			title: translate( 'View Drive' ),
+		},
+		{
+			href: getGoogleSheetsUrl( domainName ),
+			image: googleSheetsIcon,
+			imageAltText: translate( 'Google Sheets icon' ),
+			title: translate( 'View Sheets' ),
+		},
+		{
+			href: getGoogleSlidesUrl( domainName ),
+			image: googleSlidesIcon,
+			imageAltText: translate( 'Google Slides icon' ),
+			title: translate( 'View Slides' ),
+		},
+	];
+};
+
+const getMenuItems = ( domain, translate ) => {
+	if ( hasTitanMailWithUs( domain ) ) {
+		return getTitanMenuItems( translate );
+	}
+
+	if ( hasGSuiteWithUs( domain ) ) {
+		return getGmailMenuItems( domain.name, translate );
+	}
+
+	return null;
+};
+
+const ActionMenu = ( { domain } ) => {
+	const translate = useTranslate();
+	const menuItems = getMenuItems( domain, translate );
+	if ( ! menuItems ) {
+		return null;
+	}
+	return (
+		<EllipsisMenu
+			popoverClassName="email-plan-mailboxes-list__mailbox-action-menu-popover"
+			className="email-plan-mailboxes-list__mailbox-action-menu"
+		>
+			{ menuItems.map( function ( menuItem ) {
+				const { image, imageAltText, href, title } = menuItem;
```

Comment:
style nit: destructuring `menuItem` in the argument list would make the callback more concise.

Reply:
Done